rust/chains/abacus-ethereum/src/retrying.rs

```diff
@@ -94,22 +109,46 @@ where
 
                 match fut.await {
                     Ok(res) => return Ok(res),
-                    Err(e) => {
+                    Err(HttpClientError::ReqwestError(e)) => {
+                        warn!(
+                            backoff_ms,
+                            retries_remaining = self.max_requests - i - 1,
+                            error = %e,
+                            method = %method,
+                            "ReqwestError in retrying provider",
+                        );
+                        last_err = Some(HttpClientError::ReqwestError(e));
+                    }
+                    Err(HttpClientError::JsonRpcError(e)) => {
+                        // This is a client error so we do not want to retry on it.
                         warn!(
-                            backoff_seconds,
+                            backoff_ms,
                             retries_remaining = self.max_requests - i - 1,
                             error = %e,
                             method = %method,
-                            "Error in retrying provider",
+                            "JsonRpcError",
+                        );
+                        return Err(RetryingProviderError::JsonRpcClientError(
+                            HttpClientError::JsonRpcError(e),
+                        ));
+                    }
+                    Err(HttpClientError::SerdeJson { err, text }) => {
```

Comment:
Do we want to retry on a deserialization error? Seems like that'd also be a client error no?

Reply:
Honestly not sure, I didn't see any examples of it, so I was erring on the side of least change to behavior.